app/controllers/pieces_controller.rb

```diff
@@ -1,20 +1,32 @@
 require 'pry'
 class PiecesController < ApplicationController
 	before_action :select_pc, :only => [:show, :update]
+	before_action :only => :update do 
+		valid_move?(:x_position, :y_position)
+	end
 
 	def show
 		@pieces = select_pc.game.pieces
 	end
 
 	def update
-		row = params[:y_position]
-		col = params[:x_position]
+		row = params[:y_position].to_i
+		col = params[:x_position].to_i
 		@moved_pc = select_pc.update_attributes(y_position: row, x_position: col)
 		redirect_to game_path(select_pc.game.id)
 	end
 
 	private
 
+	def valid_move?(x_position, y_position)
```

Comment:
Yep, good catch with the `?` I had intended on doing this a little differently thus the `?` But it should not be included here. I will change the name as well to make it more clear. Thanks.